pkg/ipam/metadata/manager.go

```diff
@@ -95,7 +97,7 @@ func splitK8sPodName(owner string) (namespace, name string, ok bool) {
 	return namespace, name, true
 }
 
-func (m *Manager) GetIPPoolForPod(owner string) (pool string, err error) {
+func (m *Manager) GetIPPoolForPod(owner string, family ipam.Family) (pool string, err error) {
 	if m.namespaceStore == nil || m.podStore == nil {
 		return "", &ManagerStoppedError{}
 	}
```

Comment:
I think we should add a check here to reject invalid IP families, as otherwise it will silently fall back on the default pool which hides this error.


```suggestion
func (m *Manager) GetIPPoolForPod(owner string, family ipam.Family) (pool string, err error) {
	if m.namespaceStore == nil || m.podStore == nil {
		return "", &ManagerStoppedError{}
	}
	
	if family != ipam.IPv6 && family != ipam.IPv4 {
		return "", fmt.Errorf("invalid IP family: %s", family)
	}
```

Reply:
Alright, thank you for the guidance. I will add the checks.